plugins/builds/index.js

```diff
@@ -1139,360 +665,291 @@ const buildsPlugin = {
-                    !joinListNames.includes(`sd@${pipelineId}:${currentJobName}`);
 
-                // Handle no-join case
+                // Handle no-join case. Sequential Workflow
                 // Note: current job can be "external" in nextJob's perspective
-                /* CREATE AND START NEXT BUILD IF ALL 3 SCENARIOS ARE TRUE
+                /* CREATE AND START NEXT BUILD IF ALL 2 SCENARIOS ARE TRUE
                  * 1. No join
                  * 2. ([~D,B,C]->A) currentJob=D, nextJob=A, joinList(A)=[B,C]
                  *    joinList doesn't include D, so start A
-                 * 3. ([~D,B,C]-> sd@123:A) currentJob=D, nextJob=sd@123:A, joinList(A)=[sd@111:B,sd@111:C]
-                 *    joinList doesn't include sd@111:D, so start A
                  */
-                if (joinListNames.length === 0 || currentJobNotInJoinList) {
-                    // Next build is internal
-                    if (!isExternal) {
-                        const internalBuildConfig = {
-                            jobFactory,
-                            buildFactory,
+                const isORTrigger = !joinListNames.includes(current.job.name);
+
+                if (joinListNames === 0 || isORTrigger) {
+                    const internalBuildConfig = {
+                        jobFactory,
+                        buildFactory,
+                        pipelineId: current.pipeline.id,
+                        jobName: nextJobName,
+                        username,
+                        scmContext,
+                        event: current.event, // this is the parentBuild for the next build
+                        baseBranch: current.event.baseBranch || null,
+                        parentBuilds,
+                        parentBuildId: current.build.id
+                    };
+
+                    return createInternalBuild(internalBuildConfig);
+                }
+
+                // Get finished internal builds from event
+
+                logger.info(`Fetching finished builds for event ${event.id}`);
+                let finishedInternalBuilds = await getFinishedBuilds(current.event, buildFactory);
+
+                if (current.event.parentEventId) {
+                    // FIXME: On restart cases parentEventId should be fetched
+                    // from first event in the group
+                    const parallelBuilds = await getParallelBuilds({
+                        eventFactory,
+                        parentEventId: current.event.parentEventId,
+                        pipelineId: current.pipeline.id
+                    });
+
+                    finishedInternalBuilds = finishedInternalBuilds.concat(parallelBuilds);
+                }
+                fillParentBuilds(parentBuilds, current, finishedInternalBuilds);
+                // If next build is internal, look at the finished builds for this event
+                const jobId = joinObj[nextJobName].id;
```

Comment:
is this next job's id?
```suggestion
                const nextJobId = joinObj[nextJobName].id;
```